client/src/components/geneExpression/gene.tsx

```diff
@@ -83,29 +81,57 @@ class Gene extends React.Component<{}, State> {
     });
   };
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  handleDeleteGeneFromSet = () => {
+  handleDeleteGeneFromSet = (): void => {
     // @ts-expect-error ts-migrate(2339) FIXME: Property 'dispatch' does not exist on type 'Readon... Remove this comment to see the full error message
     const { dispatch, gene, geneset } = this.props;
     dispatch(actions.genesetDeleteGenes(geneset, [gene]));
   };
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  render() {
+  handleDisplayGeneInfo = async (): Promise<void> => {
+    track(EVENTS.EXPLORER_GENE_INFO_BUTTON_CLICKED);
+    // @ts-expect-error ts-migrate(2339) FIXME: Property 'dispatch' does not exist on type 'Readon... Remove this comment to see the full error message
+    const { dispatch, gene, geneId } = this.props;
+    dispatch({
+      type: "load gene info",
+      gene,
+    });
+
+    const info = await actions.fetchGeneInfo(geneId);
+    if (!info) {
+      dispatch({
+        type: "open gene info",
+        gene,
+        url: "",
+        name: "failed",
```

Comment:
Instead of overloading the name attribute, maybe we should add an error attribute that accepts null or a string